### ISSUE TITLE
Incorrect use of $reshook variable in showOptionals function in card.php for fourn/facture module

### ISSUE THREAD

**Issue by Daviid-P** [Dolibarr/dolibarr#24644]
### Bug

The showOptionals function in the card.php file of the fourn/facture directory does not properly handle the formObjectOptions hook when printing extrafields inputs.

On https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/compta/facture/card.php#L3773-L3788
we see:
```php
// Other attributes
$parameters = array('objectsrc' => !empty($objectsrc) ? $objectsrc : 0, 'colspan' => ' colspan="2"', 'cols' => '2', 'socid'=>$socid);
$reshook = $hookmanager->executeHooks('formObjectOptions', $parameters, $object, $action); // Note that $action and $object may have been modified by hook
print $hookmanager->resPrint;
if (empty($reshook)) {
    if (!empty($conf->global->THIRDPARTY_PROPAGATE_EXTRAFIELDS_TO_INVOICE) && !empty($soc->id)) {
        // copy from thirdparty
        $tpExtrafields = new Extrafields($db);
        $tpExtrafieldLabels = $tpExtrafields->fetch_name_optionals_label($soc->table_element);
        if ($soc->fetch_optionals() > 0) {
            $object->array_options = array_merge($object->array_options, $soc->array_options);
        }
    }

    print $object->showOptionals($extrafields, 'create', $parameters);
}
```
Here we execute the hook formObjectOptions to allow for any modification to the extrafields inputs that will be printed.

This works fine.

However in https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fourn/facture/card.php
first we print the extrafields inputs in https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fourn/facture/card.php#L2655-L2657
Like so:
```php
if (empty($reshook)) {
    print $object->showOptionals($extrafields, 'create');
}
```

Then we execute the hook formObjectOptions in https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fourn/facture/card.php#L2727-L2730
Like so:

```php
// Other options
$parameters = array();
$reshook = $hookmanager->executeHooks('formObjectOptions', $parameters, $object, $action); // Note that $action and $object may have been modified by hook
print $hookmanager->resPrint;
```

This is a problem because when printing showOptionals we check empty($reshook) and this can be -1 or 1 comming all the way from the first hook doActions.

In my case an extrafield that I want to force users to input but want to still allow it to be NULLable in the DB because we load data from an outside source.

### Environment Version

16.0.3

### Expected and actual behavior

To address the issue, the `formObjectOptions` hook should be executed before printing the extrafields inputs in `fourn/facture/card.php`, similar to how it is done in `compta/facture/card.php`. This will ensure that the reshook value is correctly checked when we do `showOptionals`.

So basically move https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fourn/facture/card.php#L2727-L2730 to right before https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fourn/facture/card.php#L2655-L2657

### Steps to reproduce the behavior

Return -1 from a `doAction` function in `actions_myModule.class.php`
If your invoice has extrafields you'll see they're not printed.
